### DISCUSSION TITLE
nixos system administration

### DISCUSSION THREAD

**Discussion by aleeusgr** [aleeusgr/nixos-config#3] [General]
Researching this repo for [migrating my system to flakes](https://github.com/aleeusgr/nix-things/issues/16)

**Comment by aleeusgr** [comment]
https://github.com/aleeusgr/nixos-config/blob/53e056d2fead73d4f268dbe7dfb6d0921e81592d/flake.nix#L42-L49

https://github.com/aleeusgr/nixos-config/blob/53e056d2fead73d4f268dbe7dfb6d0921e81592d/flake.nix#L51-L62

**Reply by NobbZ** [reply]
I recently switched to [`flake-parts`](https://github.com/hercules-ci/flake-parts) as kind of framework to manage my flakes, roughly as if they were configuration.

Therefore this flake currently still contains a lot of quick legacy stuff for easier migration. Basically everything you see here in the second code block is such legacy code.

The interesting part is the `home_configs.nix` and `system_config,nix` which define "options" that make it easier to create actual nixos and HM configurations as required by the respectve tooling.

Both modules are written in a way, that they follow a strict convention but make it easily possible to override specifics if necessary.

**Comment by aleeusgr** [comment]
https://github.com/aleeusgr/nixos-config/blob/53e056d2fead73d4f268dbe7dfb6d0921e81592d/parts/system_configs.nix#L8

Supplied my hardware-configuration and configuration [here](https://github.com/aleeusgr/nixos-config/blob/53e056d2fead73d4f268dbe7dfb6d0921e81592d/parts/system_configs.nix#L79-L111)

**Comment by aleeusgr** [comment]
https://github.com/aleeusgr/nixos-config/blob/53e056d2fead73d4f268dbe7dfb6d0921e81592d/parts/home_configs.nix#L86-L92
